### ISSUE TITLE
mypy linter is available but not enabled

### ISSUE THREAD

**Issue by timjstewart** [dense-analysis/ale#2956]
## Information

**VIM version**

NVIM v0.5.0-nightly-386-gf79369d42
Build type: Release

Operating System: macOS Mojave, version 10.14.6

## What went wrong

I have some python code that has some mypy type-checking errors in it.  mypy 0.761 is installed and I can run it from vim using `:!mypy`.  `:ALEInfo` lists mypy as being in the list of Available Linters but not in the list of Enabled Linters.  Editing my file with type errors in it shows other PEP8 errors but not the type errors.

If I open up ALE's `linter.vim` and explicitly add "mypy" to `l:buffer_ale_linters` in the `GetLinterNames` function, then `mypy` flags the type errors perfectly. 

## Reproducing the bug

1. Make sure mypy is installed and that vim can execute it (e.g. `:!mypy`)
2. In a python file enter the following code:

```
def checkme() -> int:
    """docstring"""
    return "not an int"
```
3. Save the file.

Expected Behavior: You should see an error on the line with the `return` statement.
Actual Behavior: I see no error on any of the lines.

Thank you!

### :ALEInfo
```
 Current Filetype: python
Available Linters: ['bandit', 'flake8', 'mypy', 'prospector', 'pycodestyle', 'pydocstyle', 'pyflakes', 'pylama', 'pylint', 'pyls', 'pyre', 'vulture']
  Enabled Linters: ['flake8', 'pylint']
 Suggested Fixers: 
  'add_blank_lines_for_python_control_statements' - Add blank lines before control statements.
  'autopep8' - Fix PEP8 issues with autopep8.
  'black' - Fix PEP8 issues with black.
  'isort' - Sort Python imports with isort.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'reorder-python-imports' - Sort Python imports with reorder-python-imports.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'yapf' - Fix Python files with yapf.
 Linter Variables:

let g:ale_python_auto_pipenv = 0
let g:ale_python_flake8_auto_pipenv = 0
let g:ale_python_flake8_change_directory = 1
let g:ale_python_flake8_executable = 'flake8'
let g:ale_python_flake8_options = ''
let g:ale_python_flake8_use_global = 0
let g:ale_python_pylint_auto_pipenv = 0
let g:ale_python_pylint_change_directory = 1
let g:ale_python_pylint_executable = 'pylint'
let g:ale_python_pylint_options = ''
let g:ale_python_pylint_use_global = 0
let g:ale_python_pylint_use_msg_id = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let b:ale_fixers = ['autopep8', 'yapf']
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let b:ale_linters = ['flake8', 'pylint']
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
let b:ale_warn_about_trailing_whitespace = 0
  Command History:

(executable check - success) flake8
(finished - exit code 0) ['/usr/local/bin/zsh', '-ic', '''flake8'' --version']

<<<OUTPUT STARTS>>>
3.7.9 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.1) CPython 3.6.9 on Darwin
<<<OUTPUT ENDS>>>

(executable check - success) pylint
(finished - exit code 16) ['/usr/local/bin/zsh', '-ic', 'cd ''/Users/STEWTJ3/src/project/math-service'' && ''pylint''  --output-format text --msg-template="{path}:{line}:{column}: {msg_id} ({symbol}) {msg}" --reports n ''/Users/STEWTJ3/src/project/math-service/math_engine/graph.py''']

<<<OUTPUT STARTS>>>
************* Module math_engine.graph
math_engine/graph.py:1:0: C0111 (missing-docstring) Missing module docstring

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)

<<<OUTPUT ENDS>>>

(finished - exit code 0) ['/usr/local/bin/zsh', '-ic', 'cd ''/Users/STEWTJ3/src/project/math-service/math_engine'' && ''flake8'' --format=default --stdin-display-name ''/Users/STEWTJ3/src/project/math-service/math_engine/graph.py'' - < ''/var/folders/lx/2hcmvn254q7cdzt_2wwm1pgc0000gp/T/nvimUjQxrk/2/graph.py''']

<<<NO OUTPUT RETURNED>>>
```

**Comment by timjstewart** [comment]
I had failed to include mypy in my ftplugin/python.vim file (I'm using NeoVim).
